sim/pokemon.ts

```diff
@@ -538,20 +538,20 @@ export class Pokemon {
 			);
 		}
 		if (adjacentOnly) allies = allies.filter(ally => this.battle.isAdjacent(this, ally));
-		return allies.filter(ally => ally && !ally.fainted);
+		return forTargeting ? allies : allies.filter(ally => ally && !ally.fainted);
 	}
 
-	foes(adjacentOnly?: boolean): Pokemon[] {
+	foes(adjacentOnly?: boolean, forTargeting?: boolean): Pokemon[] {
 		let foes = this.side.foe.active;
 		if (this.battle.gameType === 'multi') {
-			const team = this.side.foe.n % 2;
+			const team = +!(this.side.n % 2);
 			// @ts-ignore
 			foes = this.battle.sides.flatMap(side =>
 				side.n % 2 === team ? side.active : []
 			);
 		}
 		if (adjacentOnly) foes = foes.filter(foe => this.battle.isAdjacent(this, foe));
-		return foes.filter(foe => foe && !foe.fainted);
+		return forTargeting ? foes : foes.filter(foe => foe && !foe.fainted);
 	}
```

Comment:
It makes a lot of sense for `adjacentFoes` to be its own function, rather than a boolean on `foes`:

```ts
	adjacentFoes() {
		return this.foes().filter(foe => this.battle.isAdjacent(this, foe));
	}
```

Reply:
(Ditto `allies`.)

Reply:
Here's your "nicer alternative to string codes and switches".

Reply:
I _guess_

Reply:
This is a good general principle, by the way: Consider whether boolean parameters make sense as separate functions.